backend/src/memos/memos.service.ts

```diff
@@ -17,4 +24,46 @@ export class MemosService {
   async listMemosForUser(): Promise<void> {
     return
   }
+
+  // Retrieves an existing memo based on slug
+  async getMemo(slug: string): Promise<GetMemoResponseDto> {
```

Comment:
TODO: Different responses for void or expired memos?

Reply:
I think this is a frontend thing. Void and expired memos should still be viewable but explicitly (i.e. visually) marked as void/expired.